### PR TITLE
Prevent accidental double-recovery

### PR DESCRIPTION
**Pull request by bfollington**

### CODE REVIEW

File: xcode/Subconscious/Shared/Components/Common/Recovery/RecoveryModeFormPanelView.swift [hunk 1]
```diff
@@ -75,8 +75,10 @@ struct RecoveryModeFormPanelView: View {
             
             Button(
                 action: {
+                    // Dismiss recovery if already succeeded
                     if store.state.recoveryStatus == .succeeded {
                         store.send(.requestPresent(false))
+                        return
```
Comment (gordonbrander) [inline xcode/Subconscious/Shared/Components/Common/Recovery/RecoveryModeFormPanelView.swift:81]: LOL 🤦 

One way we could be more sure we aren't hitting cases like this is to factor this code out into an update function. Say `.recoveryButtonPressed`. In the `recoveryButtonPressed` update function we could perform this same logic.

The advantage would be that we could create a unit test for the update logic to make very sure we don't run recovery twice.

**Review by gordonbrander — APPROVED** [review]
Approved, but with a suggestion to improve further.

File: xcode/Subconscious/Shared/Components/Common/Recovery/RecoveryModeFormPanelView.swift (hunk 1 above)
Comment (gordonbrander) [reply xcode/Subconscious/Shared/Components/Common/Recovery/RecoveryModeFormPanelView.swift:81]: I've created a PR to implement this suggestion, along with tests https://github.com/subconsciousnetwork/subconscious/pull/917

Comment (bfollington) [reply xcode/Subconscious/Shared/Components/Common/Recovery/RecoveryModeFormPanelView.swift]: Merged your suggestions into my PR, definitely a better approach.

**Pull request merged**